<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cubiomes Viewer</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/icons/map.png</normaloff>:/icons/map.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">QToolButton:checked { background-color:rgb(85, 87, 83) }
QToolButton:!checked { background-color:rgb(240, 240, 240) }</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_6">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitterMap">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QFrame" name="controlFrame">
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QGridLayout" name="gridMCSeed">
          <item row="0" column="3">
           <widget class="QLineEdit" name="seedEdit">
            <property name="toolTip">
             <string>Press enter to accept</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="labelSeed">
            <property name="toolTip">
             <string>Seed can be an integer or text. Leave empty for a random seed</string>
            </property>
            <property name="text">
             <string>seed:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QLabel" name="labelSeedType">
            <property name="minimumSize">
             <size>
              <width>61</width>
              <height>25</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Seed can be an integer or text. Leave empty for a random seed</string>
            </property>
            <property name="text">
             <string>(random)</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labeMC">
            <property name="toolTip">
             <string>Minecraft version</string>
            </property>
            <property name="text">
             <string>MC</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="comboBoxMC">
            <property name="toolTip">
             <string>Minecraft version</string>
            </property>
            <item>
             <property name="text">
              <string>1.16</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.15</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.14</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.13</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.12</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.11</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.10</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.9</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.8</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.7</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTabWidget" name="tabContainer">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tabSearch">
           <attribute name="title">
            <string>Search</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QSplitter" name="splitterSearch">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <widget class="QGroupBox" name="groupConstraints">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="title">
                <string>Constraints</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_2">
                <property name="leftMargin">
                 <number>4</number>
                </property>
                <property name="topMargin">
                 <number>4</number>
                </property>
                <property name="rightMargin">
                 <number>4</number>
                </property>
                <property name="bottomMargin">
                 <number>4</number>
                </property>
                <item row="3" column="0">
                 <widget class="QPushButton" name="buttonInfo">
                  <property name="maximumSize">
                   <size>
                    <width>20</width>
                    <height>16777215</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>?</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QPushButton" name="buttonRemoveAll">
                  <property name="text">
                   <string>Remove all</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0" colspan="5">
                 <widget class="QSplitter" name="splitterConditions">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <widget class="QListWidget" name="listConditions48">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>20</height>
                    </size>
                   </property>
                   <property name="font">
                    <font>
                     <family>Monospace</family>
                    </font>
                   </property>
                   <property name="toolTip">
                    <string>Conditions for the 48-bit seed generator</string>
                   </property>
                   <property name="dragDropMode">
                    <enum>QAbstractItemView::InternalMove</enum>
                   </property>
                   <property name="selectionMode">
                    <enum>QAbstractItemView::ExtendedSelection</enum>
                   </property>
                   <property name="movement">
                    <enum>QListView::Snap</enum>
                   </property>
                  </widget>
                  <widget class="QListWidget" name="listConditionsFull">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>20</height>
                    </size>
                   </property>
                   <property name="font">
                    <font>
                     <family>Monospace</family>
                    </font>
                   </property>
                   <property name="toolTip">
                    <string>Conditions with full biome dependency</string>
                   </property>
                   <property name="dragDropMode">
                    <enum>QAbstractItemView::InternalMove</enum>
                   </property>
                   <property name="selectionMode">
                    <enum>QAbstractItemView::ExtendedSelection</enum>
                   </property>
                   <property name="movement">
                    <enum>QListView::Snap</enum>
                   </property>
                  </widget>
                 </widget>
                </item>
                <item row="3" column="3">
                 <widget class="QPushButton" name="buttonEdit">
                  <property name="text">
                   <string>Edit</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="2">
                 <widget class="QPushButton" name="buttonRemove">
                  <property name="text">
                   <string>Remove selection</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="4">
                 <widget class="QPushButton" name="buttonAddFilter">
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
              <widget class="QGroupBox" name="groupMatching">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="title">
                <string>Matching seeds</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_3">
                <property name="leftMargin">
                 <number>4</number>
                </property>
                <property name="topMargin">
                 <number>4</number>
                </property>
                <property name="rightMargin">
                 <number>4</number>
                </property>
                <property name="bottomMargin">
                 <number>4</number>
                </property>
                <item row="0" column="0">
                 <widget class="QTableWidget" name="listResults">
                  <property name="font">
                   <font>
                    <family>Monospace</family>
                   </font>
                  </property>
                  <property name="contextMenuPolicy">
                   <enum>Qt::CustomContextMenu</enum>
                  </property>
                  <property name="horizontalScrollBarPolicy">
                   <enum>Qt::ScrollBarAlwaysOff</enum>
                  </property>
                  <property name="editTriggers">
                   <set>QAbstractItemView::NoEditTriggers</set>
                  </property>
                  <property name="selectionMode">
                   <enum>QAbstractItemView::SingleSelection</enum>
                  </property>
                  <property name="selectionBehavior">
                   <enum>QAbstractItemView::SelectRows</enum>
                  </property>
                  <property name="sortingEnabled">
                   <bool>true</bool>
                  </property>
                  <attribute name="horizontalHeaderDefaultSectionSize">
                   <number>160</number>
                  </attribute>
                  <attribute name="horizontalHeaderHighlightSections">
                   <bool>false</bool>
                  </attribute>
                  <attribute name="horizontalHeaderStretchLastSection">
                   <bool>true</bool>
                  </attribute>
                  <attribute name="verticalHeaderDefaultSectionSize">
                   <number>20</number>
                  </attribute>
                  <column>
                   <property name="text">
                    <string>Hex  (Low-48 | Top-16)</string>
                   </property>
                   <property name="textAlignment">
                    <set>AlignLeading|AlignVCenter</set>
                   </property>
                  </column>
                  <column>
                   <property name="text">
                    <string>Seed</string>
                   </property>
                   <property name="textAlignment">
                    <set>AlignLeading|AlignVCenter</set>
                   </property>
                  </column>
                 </widget>
                </item>
                <item row="1" column="0">
                 <layout class="QGridLayout" name="gridLayout">
                  <item row="0" column="4">
                   <widget class="QPushButton" name="buttonLoadList">
                    <property name="enabled">
                     <bool>false</bool>
                    </property>
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <property name="maximumSize">
                     <size>
                      <width>24</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>...</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="2" colspan="2">
                   <widget class="QComboBox" name="comboSearchType">
                    <item>
                     <property name="text">
                      <string>incremental</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>48-bit family blocks</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>seed list from a file...</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item row="0" column="6">
                   <widget class="QSpinBox" name="spinThreads">
                    <property name="minimum">
                     <number>1</number>
                    </property>
                    <property name="maximum">
                     <number>16</number>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="0">
                   <widget class="QLabel" name="label">
                    <property name="toolTip">
                     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;searching algorithm&lt;/p&gt;&lt;p&gt;incremental: best for non-exhaustive biome searches&lt;/p&gt;&lt;p&gt;48-bit family blocks: best for searches with strong structure restrictions&lt;/p&gt;&lt;p&gt;seed list from file...: load and search through a pre-computed list of seed&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                    </property>
                    <property name="text">
                     <string>Search:</string>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="0" colspan="7">
                   <widget class="QProgressBar" name="progressBar">
                    <property name="font">
                     <font>
                      <family>Monospace</family>
                     </font>
                    </property>
                    <property name="toolTip">
                     <string>Progress within the set of all 48-bit seeds.</string>
                    </property>
                    <property name="layoutDirection">
                     <enum>Qt::LeftToRight</enum>
                    </property>
                    <property name="maximum">
                     <number>10000</number>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignCenter</set>
                    </property>
                    <property name="format">
                     <string>0</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QPushButton" name="buttonSearchHelp">
                    <property name="maximumSize">
                     <size>
                      <width>20</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>?</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>Seed:</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="4" colspan="3">
                   <widget class="QCheckBox" name="checkStop">
                    <property name="toolTip">
                     <string>stop as soon as the next set of matching seeds is found</string>
                    </property>
                    <property name="text">
                     <string>Stop on results</string>
                    </property>
                    <property name="checked">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="5">
                   <widget class="QLabel" name="label_3">
                    <property name="text">
                     <string>Threads:</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="2" colspan="2">
                   <widget class="QLineEdit" name="lineStart">
                    <property name="text">
                     <string>0</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="0" colspan="7">
                   <layout class="QHBoxLayout" name="horizontalLayout">
                    <item>
                     <widget class="QPushButton" name="buttonClear">
                      <property name="text">
                       <string>Clear results</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QPushButton" name="buttonStart">
                      <property name="text">
                       <string>Start search</string>
                      </property>
                      <property name="checkable">
                       <bool>true</bool>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frameMap">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="MapView" name="mapView" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMap">
    <property name="title">
     <string>Map</string>
    </property>
    <addaction name="actionGo_to"/>
    <addaction name="actionScan_seed_for_Quad_Huts"/>
    <addaction name="actionOpen_shadow_seed"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="actionLoad"/>
    <addaction name="separator"/>
    <addaction name="actionSearch_seed_list"/>
    <addaction name="actionSearch_full_seed_space"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuMap"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionGo_to">
   <property name="text">
    <string>Go to...</string>
   </property>
  </action>
  <action name="actionScan_seed_for_Quad_Huts">
   <property name="text">
    <string>Scan seed for Quad-Huts...</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionOpen_shadow_seed">
   <property name="text">
    <string>Open shadow seed</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save progress...</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load progress...</string>
   </property>
  </action>
  <action name="actionSearch_seed_list">
   <property name="text">
    <string>Search seed list...</string>
   </property>
   <property name="toolTip">
    <string>Load seeds from file for search</string>
   </property>
  </action>
  <action name="actionSearch_full_seed_space">
   <property name="text">
    <string>Search full seed space</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy seeds from list</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>Paste seeds into list</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences...</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MapView</class>
   <extends>QWidget</extends>
   <header>mapview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections/>
</ui>
